x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldTests.java

```diff
@@ -267,30 +295,41 @@ public static Object randomSemanticTextInput() {
         }
     }
 
-    public static ChunkedInferenceServiceResults toChunkedResult(SemanticTextField field) throws IOException {
+    public static ChunkedInferenceServiceResults toChunkedResult(Map<String, List<String>> matchedTextMap, SemanticTextField field)
+        throws IOException {
         switch (field.inference().modelSettings().taskType()) {
             case SPARSE_EMBEDDING -> {
                 List<MlChunkedTextExpansionResults.ChunkedResult> chunks = new ArrayList<>();
                 for (var entry : field.inference().chunks().entrySet()) {
-                    for (var chunk : entry.getValue()) {
+                    String entryField = entry.getKey();
+                    List<SemanticTextField.Chunk> entryChunks = entry.getValue();
+                    List<String> entryFieldMatchedText = validateAndGetMatchedTextForField(matchedTextMap, entryField, entryChunks.size());
+
+                    ListIterator<String> matchedTextIt = entryFieldMatchedText.listIterator();
```

Comment:
Nitpick: Can we use either `Iterator` or `Iter` instead of `It`? I find the name slightly confusing to read.